src/Mailgun/Connection/RestClient.php

```diff
@@ -147,6 +147,9 @@ public function post($endpointUrl, array $postData = [], $files = [])
                     $postDataMultipart[] = [
                         'name' => sprintf('%s[%d]', $key, $index++),
                         'contents' => $subValue,
+                        'headers' => [
+                            'Content-Disposition' => sprintf('form-data; name="%s"', $key),
```

Comment:
As far as I understood, the automatic decoding of square brackets in form names is a specialty of PHP. The applicable [RFC 7578](https://tools.ietf.org/html/rfc7578) does not impose such constraints on names nor does it advise to use brackets to encode arrays.